rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -85,23 +83,8 @@ def import_rmw_implementation():
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
     available_implementations = get_rmw_implementations()
-    if __selected_rmw_implementation is None:
-        logger = logging.getLogger('rclpy')
-        # prefer FastRTPS, otherwise first in alphabetical order
-        # the same logic is implemented in
-        # rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake
-        default_impl = 'rmw_fastrtps_cpp'
-        if default_impl in available_implementations:
-            select_rmw_implementation(default_impl)
-        elif available_implementations:
-            select_rmw_implementation(available_implementations[0])  # select the first one
-        else:
-            raise NoImplementationAvailableException()
-        logger.debug("Implicitly selecting the '{0}' rmw implementation."
-                     .format(__selected_rmw_implementation))
-    module_name = '._rclpy__{rmw_implementation}'.format(
-        rmw_implementation=__selected_rmw_implementation,
-    )
+    __selected_rmw_implementation = 'rmw_implementation'
```

Comment:
> You're right, though I suppose you mean `rcl` not `rclcpp`. You'll have to either remove the code and tests for the Python selection of rmw impl (as is proposed in this pr) or fix those functions/features to use the new mechanism, assuming that is possible.

As mentioned above I have already removed these in this PR in 9e0597807ad9b64397ba35eaa549f569978a039e.

> Doesn't it also mean that all of the `get_X_information` macros (where X is `rcl`, `rclcpp`, etc.) are no longer useful and should be removed? If that is the case, then that is another example of dead code that should be removed.

Yes, various CMake function can be removed. All if this is being tracked in the parent ticket: ros2/ros2#302.

Again I don't see the point of squeezing as much changes as possible into this PR set. Especially since the previous rounds of PRs have moved very fast. It takes a lot of time to get these changes into the state to be "green". Imo this discussion is slowing the progress more down then if this would have been merged. Then the remaining tasks could have already been started / done by now.

I can remove the CMake functions in the corresponding PR if that makes a difference for you.